ServerCore/PuzzleStateHelper.cs

```diff
@@ -569,6 +569,14 @@ public static async Task UpdateTeamsWhoSentResponse(PuzzleServerContext context,
 
                     await context.SaveChangesAsync();
                     transaction.Commit();
+
```

Comment:
The query above and possibly below need to ensure that the team's submission was for this puzzle. Otherwise we get a bug where if a team submitted "foo" for puzzle A, and puzzle B adds "foo" as a response, this logic triggers for puzzle B

Reply:
Indeed!